### PR TITLE
Rehierl issue 912

### PR DESCRIPTION
**Pull request by rehierl**
These are changes mostly related to issue #912

### .multipage-split.sh

Without the removal of the tools folder, that script can only be used a single time. It will fail if you try to execute it a second time and that tools folder exists and is not empty. I can't tell if that is intentional?

### docs/build-documentation.md

I guess Node.js's URL got removed after quite a few developers left that project. Now, as they seem reunited, that URL could be reinserted?

### sections/semantics-sections.include

(line numbers refer to the new numbering; including the changes)

(1) lines 1258 to 1261
(2) line 1258 - i.e. "explicit"
(3) lines 1354 to 1357
(4) lines 1379 to 1397
(5) line 1405
(6) lines 1439 to 1570

#### about (2)

I assume that each sectioning content element will start a new (it's own) explicit section. What is still unclear to me is what sectioning roots will do; see lines 1268 (original numbering). Will they represent a separate explicit section, or will they appear as any other element (like a &lt;p&gt; element for example)?

#### about lines 1551 to 1554

You could translate these lines into the following pseudocode:

```js
outline = currentOwner.outline
heading = outline.lastSection.heading

//- read as: if(A or B)
//- node is the heading node that is entered
if(heading.implied || (node.rank() >= heading.rank()))
  //- execute the steps described
```

It should be clear that you can't determine the rank of an implied heading, because it does not correspond to an existing heading element. Then keep in mind how such an if statement will be executed: if(A or B) will first evaluate A. If A evaluates to 'true', then B will not be evaluated. If A evaluates to 'false', then B will be evaluated next.

In other words: If you take the original order, you would first try to determine the rank of a heading. This would cause an error if that heading was an implied heading.

### CODE REVIEW

**Review by travisleithead — APPROVED** [review]
All these changes LGTM.

**Comment by rehierl** [comment]
I forgot to mention that I have added the pseudocode for the 'tree order' (4) using the example pattern. I don't know if that is fine with you, because these examples usually only contain HTML fragments. So this might be a little irritating for readers. It could be worth it to use another pattern, if there is one?

**Comment by chaals** [comment]
Marked as non-substantive for IPR from ash-nazg.

**Comment by rehierl** [comment]
@chaals - I don't feel that I have contributed enough to be mentioned in the acknowledgements. But thanks for asking.

Branch deleted.

**Comment by rehierl** [comment]
After having re-read the change I made to part 2 (see below), I don't exactly see it as an error, but also not as a good improvement. I can't exactly suggest to change that part even further, because any additional change could make things even worse ...

In the hopes that this detailed explanation might help someone to put that change into perspective ...

--

[From the 5.1 spec (Recommendation)](https://www.w3.org/TR/html51/sections.html#headings-and-sections): *""(1) The first element of heading content in an element of sectioning content represents the heading for that section. **(2)** Subsequent headings of equal or higher rank start new (implied) sections, (3) headings of lower rank start implied subsections that are part of the previous one. (4) In both cases, the element represents the heading of the implied section.""*

[From the 5.2 spec (Candidate Recommendation)](https://www.w3.org/TR/html52/sections.html#headings-and-sections): *""(1) The first element of heading content in an element of sectioning content represents the heading for that explicit section. **(2)** Subsequent headings of equal or higher rank start new implied subsections that are part of the previous section’s parent section. (3) Subsequent headings of lower rank start new implied subsections that are part of the previous one. (4) In both cases, the element represents the heading of the implied section.""*

- Adding "explicit" to part (1) still makes it a bit more clear.
- Part (3) changed, but its meaning remained as is.
- Part (4) did not change at all.

What was not exactly helpful was changing part (2):

- (5.1) "Subsequent headings of equal or higher rank start new (implied) sections,"
- (5.2) "Subsequent headings of equal or higher rank start new implied subsections that are part of the previous section’s parent section."

The main reason for my change was to break up that sentence, that merged two different aspects ("headings of equal or higher rank" and "headings of lower rank") into one. I still believe that this is a tiny improvement.

The issue here are those three tiny letters - "section" versus "sub-section":

**The 5.1 version of part (2)** is a top-down view that has the following focus: If you enter a heading that has a rank which is equal or greater than the rank of any of the previous relevant headings, then add a new implied section to the outline of the first outer sectioning element. This is consistent with the
algorithm's definition ("When entering a heading content element" / "Otherwise, if the element ..." / "create a new section and append it to the outline").

So the the critical part here is, that the algorithm does *not* add such a new implied section as a *subsection* to any of the previous ones. This new implied section will become a direct or distant **sibling section** of the first inner section of a sectioning element. (The outline is defined as a list of sections.)

**This version lacks clarification of what is happening in between** - i.e. when a heading has a rank that is greater or equal than the rank of the *previous couple of headings* (not: *any of the previous relevant headings*). This case is being dealt with by ("When entering a heading content element" /
"Otherwise, run these substeps" / "Heading loop"). This is a bit difficult to spot, because if the heading's rank *is not lower*, the implied section will be added *as subsection to the candidate section's parent section*.

**The 5.2 version of part (2)** is a bottom-up view that has the following focus: If you have a heading that has a rank which is equal or greater than the rank of your current candidate section, then add the implied section as sub-section to the candidate section's parent section. To be a bit more clear, it could have been better to state "one of the previous section's ancestor sections" instead of "previous section's parent section".

**This version lacks clarification of what is happening at the top** - i.e. when a heading has a rank that is greater or equal than the rank of *any of the previous relevant headings* (not: *previous couple of headings*).

**Conclusion**: So 5.1's version explains 1 (any), but skips 2 (previous couple). And 5.2's version skips 1, but explains 2.
